prow/plugins/project/project.go

```diff
@@ -0,0 +1,257 @@
+		Config: func(repos []string) map[string]string {
+			configMap := make(map[string]string)
+			for _, repo := range repos {
+				team, exists := config.ProjectConfigMap[repo]
+				if exists {
+					configMap[repo] = msgForTeam(team)
+				}
+			}
+			configMap[""] = msgForTeam(config.ProjectConfigMap[""])
+			return configMap
+		}(enabledRepos),
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/project <board>, /project <board> <column>, or /project clear <board>",
+		Description: "Updates the project board and column for an issue or PR",
+		Featured:    false,
+		WhoCanUse:   "Members of the project maintainers GitHub team can use the '/project' command.",
+		Examples:    []string{"/project 0.5.0", "/project 0.5.0 To do", "/project clear 0.4.0"},
+	})
+	return pluginHelp, nil
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, &e, pc.PluginConfig.ProjectConfigMap)
+}
+
+func updateProjectNameToIDMap(projects []github.Project) {
+	for _, project := range projects {
+		projectNameToIDMap[project.Name] = project.ID
+	}
+}
+
+func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, projectConfigMap map[string]plugins.ProjectConfig) error {
+	if e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	matches := projectRegex.FindStringSubmatch(e.Body)
+	if len(matches) != 2 && len(matches) != 3 {
+		return fmt.Errorf(invalidNumArgs)
+	}
+
+	proposedProject := matches[1]
+	proposedColumnName := ""
+	if len(matches) > 2 && proposedProject != clearKeyword {
+		proposedColumnName = matches[2]
+	}
+	projectToClear := ""
+	if proposedProject == clearKeyword {
+		projectToClear = matches[2]
+	}
+
+	// See if the current user is part of the team in order to set the project/column
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	projectMaintainers, err := gc.ListTeamMembers(projectConfigMap[repo].ProjectMaintainersID, github.RoleAll)
+	if err != nil {
+		return err
+	}
+	found := false
+	for _, person := range projectMaintainers {
+		login := github.NormLogin(e.User.Login)
+		if github.NormLogin(person.Login) == login {
+			found = true
+			break
+		}
+	}
+	if !found {
+		// not in the project maintainers team
+		msg := fmt.Sprintf(notATeamMemberMsg, org, repo, org, repo)
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	repos, err := gc.GetRepos(org, false)
+	if err != nil {
+		log.WithError(err).Errorf("Error listing all repos in org %s", org)
+		return err
+	}
+	// Get all projects for all repos
+	var projects []github.Project
+	for _, repo := range repos {
+		repoProjects, err := gc.GetRepoProjects(org, repo.Name)
+		if err != nil {
+			log.WithError(err).Errorf("Error listing all projects under repo %s", repo.Name)
+			return err
```

Comment:
In these cases we should report some internal error to the user on GitHub and ask them to re-try their command